CRM/Contribute/BAO/Contribution.php

```diff
@@ -3961,7 +3960,7 @@ public static function recordAdditionalPayment($contributionId, $trxnsData, $pay
         $contributionDetails = CRM_Core_DAO::setFieldValue('CRM_Contribute_BAO_Contribution', $contributionId, 'contribution_status_id', $statusId);
       }
       // add financial item entry
-      $financialItemStatus = CRM_Core_PseudoConstant::get('CRM_Financial_DAO_FinancialItem', 'status_id');
+      $paidStatus = CRM_Core_PseudoConstant::getKey('CRM_Financial_DAO_FinancialItem', 'status_id', 'Paid');
```

Comment:
@samuelsov can you initialise this ```$paidStatus``` outside the IF-ELSEIF loop, that could avoid the repeated assignment.